### PR TITLE
feat: add RadioGroup to UI kit

### PR DESCRIPTION
**Pull request by thiagodallacqua-hpe**
## Description

Ticket: WEB-1257

Add `RadioGroup` into the UI kit.

## Test Plan

- Go to the `/design/#RadioGroup` page

Screen Shot 2023-10-16 at 16 36 19

Screen Shot 2023-10-16 at 16 48 37

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/pages/Clusters/ClustersQueuedChart.tsx [hunk 1]
```diff
@@ -32,19 +32,29 @@ const ClustersQueuedChart: React.FC<Props> = ({ poolStats }: Props) => {
     return [now - viewDays * 86400, now];
   }, [viewDays]);
 
+  const options = [
```
Comment (keita-determined) [inline webui/react/src/pages/Clusters/ClustersQueuedChart.tsx:35]: Do we want this border? We dont have that in the current radio buttons

image

File: webui/react/src/components/kit/RadioGroup.tsx [hunk 2]
```diff
@@ -10,12 +10,14 @@ import useResize from 'hooks/useResize';
 
 import css from './RadioGroup.module.scss';
 
-interface Props {
+interface Props extends Omit<RadioGroupProps, 'onChange' | 'options'> {
```
Comment (keita-determined) [inline webui/react/src/components/kit/RadioGroup.tsx:13]: Some props are overlapped with `RadioGroupProps`

File: webui/react/src/components/kit/RadioGroup.tsx
Comment (keita-determined) [inline webui/react/src/components/kit/RadioGroup.tsx:1]: bug: 2 borders

image

**Review by keita-determined — CHANGES_REQUESTED** [review]

File: webui/react/src/components/kit/RadioGroup.tsx (hunk 2 above)
Comment (keita-determined) [reply webui/react/src/components/kit/RadioGroup.tsx:13]: Then `onChange` doesn't have to be omitted??
Anyways what i meant is that we should follow the other components. Most of UIKit components don't extend props with antd props

**Comment by ashtonG** [comment]
quick note: please remember to remove/replace the dependencies on `ConditionalWrapper` and `useResize` on the module -- these currently rely on stuff outside the UI Kit which will not be in the new repo.

File: webui/react/src/pages/Clusters/ClustersQueuedChart.tsx [hunk 3]
```diff
@@ -32,19 +30,28 @@ const ClustersQueuedChart: React.FC<Props> = ({ poolStats }: Props) => {
...
-          <Radio.Group
-            className={css.filter}
-            value={viewDays}
-            onChange={(e) => setViewDays(e.target.value)}>
-            <Radio.Button value={7}>7 days</Radio.Button>
-            <Radio.Button value={30}>30 days</Radio.Button>
-          </Radio.Group>
+          <RadioGroup
+            options={options}
+            value={String(viewDays)}
```
Comment (ashtonG) [inline webui/react/src/pages/Clusters/ClustersQueuedChart.tsx:52]: can we just allow the value to be a number or string?

**Pull request closed without merging**